src/anomalib/utils/visualization/base.py

```diff
@@ -0,0 +1,43 @@
+"""Base visualization generator."""
+
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from abc import ABC, abstractmethod
+from collections.abc import Iterator
+from dataclasses import dataclass
+from enum import Enum
+from pathlib import Path
+
+import numpy as np
+
+
+@dataclass
+class GeneratorResult:
+    """Generator result."""
```

Comment:
can you add a bit more description to these docstrings? Like where are these objects are used? How are they used etc. Just for a future reference?